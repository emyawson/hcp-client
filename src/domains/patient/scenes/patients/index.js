export * from './store';

export * from './patients.container';
