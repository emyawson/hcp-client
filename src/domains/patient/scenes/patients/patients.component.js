import React, { Component } from 'react';
import { Link } from 'react-router-dom';

export class Patients extends Component {
  componentDidMount() {
    const { getAllPatients } = this.props;

    getAllPatients();
  }
  render() {
    const { isFetchingPatients, patients, match } = this.props;
    return (
      <div>
        <h2>This is the Patients page</h2>
        <ul>
          {!isFetchingPatients &&
            patients.map(patient => (
              <li key={patient.id}>
                <Link to={`${match.url}/${patient.id}`}>
                  {`${patient.firstName} ${patient.surName}`}
                </Link>
              </li>
            ))}
        </ul>
      </div>
    );
  }
}
