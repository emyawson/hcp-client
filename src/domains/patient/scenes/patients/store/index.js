export * from './patients.reducer';
export * from './patients.epic';
