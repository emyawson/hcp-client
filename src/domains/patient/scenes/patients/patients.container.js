import { compose } from 'recompose';
import { connect } from 'react-redux';

import { mapDispatchers } from 'src/utils';
import { withNavigators } from 'src/utils/with-navigators';

import { Patients } from './patients.component';
import { patientsConnector } from './store/patients.selector';
import { getAllPatients } from './store/patients.action';

const dispatchers = mapDispatchers({
  getAllPatients,
});

export const PatientsContainer = compose(
  connect(
    patientsConnector,
    dispatchers,
  ),
  withNavigators({ hasLeftNav: true, hasTopNav: true }),
)(Patients);
