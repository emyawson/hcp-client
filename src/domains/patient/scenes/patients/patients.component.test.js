import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import { Patients } from './patients.component';

it('renders correctly', () => {
  const mockProps = {
    getAllPatients: () => {},
    isFetchingPatients: false,
    patients: [
      {
        id: 1,
        name: {
          first: 'Bob',
          last: 'Smith',
        },
        diabetesType: 1,
        birthdate: 19900101,
        phone: 212111111,
      },
      {
        id: 2,
        name: {
          first: 'Jane',
          last: 'Lee',
        },
        diabetesType: 2,
        birthdate: 19801212,
        phone: 2122222222,
      },
    ],
    match: {
      url: '/',
      isExact: false,
      params: {},
      path: '/',
    },
  };

  const component = shallow(<Patients {...mockProps} />);

  expect(component).toMatchSnapshot();
});

it('renders without crashing', () => {
  const mockProps = {
    getAllPatients: () => {},
    isFetchingPatients: false,
    patients: [],
    match: {
      url: '/',
      isExact: false,
      params: {},
      path: '/',
    },
  };

  const div = document.createElement('div');

  ReactDOM.render(<Patients {...mockProps} />, div);
});
