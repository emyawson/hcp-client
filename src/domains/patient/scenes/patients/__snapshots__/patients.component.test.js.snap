// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Patients
    getAllPatients={[Function]}
    isFetchingPatients={false}
    match={
      Object {
        "isExact": false,
        "params": Object {},
        "path": "/",
        "url": "/",
      }
    }
    patients={
      Array [
        Object {
          "birthdate": 19900101,
          "diabetesType": 1,
          "id": 1,
          "name": Object {
            "first": "Bob",
            "last": "Smith",
          },
          "phone": 212111111,
        },
        Object {
          "birthdate": 19801212,
          "diabetesType": 2,
          "id": 2,
          "name": Object {
            "first": "Jane",
            "last": "Lee",
          },
          "phone": 2122222222,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          This is the Patients page
        </h2>,
        <ul>
          <li>
            <Link
              replace={false}
              to="//1"
            >
              undefined undefined
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="//2"
            >
              undefined undefined
            </Link>
          </li>
        </ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "This is the Patients page",
        },
        "ref": null,
        "rendered": "This is the Patients page",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <Link
                replace={false}
                to="//1"
              >
                undefined undefined
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="//2"
              >
                undefined undefined
              </Link>
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="//1"
              >
                undefined undefined
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "undefined undefined",
                "replace": false,
                "to": "//1",
              },
              "ref": null,
              "rendered": "undefined undefined",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="//2"
              >
                undefined undefined
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "undefined undefined",
                "replace": false,
                "to": "//2",
              },
              "ref": null,
              "rendered": "undefined undefined",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            This is the Patients page
          </h2>,
          <ul>
            <li>
              <Link
                replace={false}
                to="//1"
              >
                undefined undefined
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="//2"
              >
                undefined undefined
              </Link>
            </li>
          </ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "This is the Patients page",
          },
          "ref": null,
          "rendered": "This is the Patients page",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <Link
                  replace={false}
                  to="//1"
                >
                  undefined undefined
                </Link>
              </li>,
              <li>
                <Link
                  replace={false}
                  to="//2"
                >
                  undefined undefined
                </Link>
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="//1"
                >
                  undefined undefined
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "undefined undefined",
                  "replace": false,
                  "to": "//1",
                },
                "ref": null,
                "rendered": "undefined undefined",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="//2"
                >
                  undefined undefined
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "undefined undefined",
                  "replace": false,
                  "to": "//2",
                },
                "ref": null,
                "rendered": "undefined undefined",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
